src/main.js

```diff
@@ -18,154 +20,307 @@ import {
   getChangesetEntriesMap,
   getChangesetFileContent,
   isSkipEntry,
+  isGitHubAppNotInstalledOrSuspended,
   formatPostComment,
 } from "./utils/index.js";
 
-async function run() {
-  // Initialize Octokit client with the GitHub token
+import { ManualChangesetCreationReminderInfo } from "./infos/index.js";
+import {
+  ChangesetFileNotAddedYetError,
+  ChangesetFileMustNotExistWithSkipEntryOption,
+} from "./errors/index.js";
+
+// ****************************************************************************
+// I) MAIN
+// ****************************************************************************
+
+const run = async () => {
   const octokit = authServices.getOctokitClient();
+  let prData = extractPullRequestData();
+
+  // If Github App is not installed or suspended, use manual approach to create changeset file
+  if (await isGitHubAppNotInstalledOrSuspended(prData)) {
+    console.log(
+      "GitHub App is not installed or suspended in the forked repository.\nProceding with checks for manual changeset creation."
+    );
+    await handleManualChangesetCreation(octokit, prData);
+  }
+  // Else, use automated approach to create changeset file
+  else {
+    console.log(
+      "GitHub App is installed and not suspended in the forked repository.\nProceding fwith checks in changelog PR description and automatic creation of changeset file."
+    );
+    await handleAutomaticChangesetCreation(octokit, prData);
+  }
+};
+
+run();
 
-  const changesetFilePath = (prNumber) => `${CHANGESET_PATH}/${prNumber}.yml`;
-  let baseOwner,
-    baseRepo,
-    baseBranch,
-    headOwner,
-    headRepo,
-    headBranch,
-    prNumber,
-    prDescription,
-    prLink;
+// ****************************************************************************
+// II) HELPERS
+// ****************************************************************************
 
+// ----------------------------------------------------------
+// Chnageset Creation Helpers Functions
+// ----------------------------------------------------------
+const handleAutomaticChangesetCreation = async (octokit, prData) => {
+  const changesetCreationMode = "automatic";
   try {
-    // Step 0 - Extract information from the payload
-    ({
-      baseOwner,
-      baseRepo,
-      baseBranch,
-      headOwner,
-      headRepo,
-      headBranch,
-      prNumber,
-      prDescription,
-      prLink,
-    } = extractPullRequestData());
-
-    // Step 1 - Parse changelog entries and validate
     const changelogEntries = extractChangelogEntries(
-      prDescription,
-      processChangelogLine
+      prData.prDescription,
+      processChangelogLine,
+      changesetCreationMode
     );
     const changesetEntriesMap = getChangesetEntriesMap(
       changelogEntries,
-      prNumber,
-      prLink
+      prData.prNumber,
+      prData.prLink,
+      changesetCreationMode
     );
-
-    // Step 2 - Handle "skip" option
-
     if (isSkipEntry(changesetEntriesMap)) {
```

Comment:
Nit: Consider pulling this up in the call tree. Both the automatic and manual flows need the changeset map. Fetch it in the beginning and pass it to each of the flows.

Reply:
Done!